lib/galaxy/model/__init__.py

```diff
@@ -10233,6 +10237,111 @@ class CleanupEventImplicitlyConvertedDatasetAssociationAssociation(Base):
     icda_id = Column(Integer, ForeignKey("implicitly_converted_dataset_association.id"), index=True)
 
 
+class CeleryUserRateLimit(Base):
+    """
+    For each user stores the last time a task was scheduled for execution.
+    Used to limit the number of tasks allowed per user per second.
+    """
+
+    __tablename__ = "celery_user_rate_limit"
+
+    user_id = Column(Integer, ForeignKey("galaxy_user.id", ondelete="CASCADE"), primary_key=True)
+    last_scheduled_time = Column(DateTime, nullable=False)
+
+    def calculate_task_start_time(
+        self, user_id: int, sa_session: Session, task_interval_secs: float, now: datetime.datetime
+    ) -> datetime.datetime:
+        """
+        Calculates the next time a task should be scheduled to run for the user_id.
+        It looks up the latest time a task has already been scheduled to run
+        for this user in a database table and then adds task_interval_secs seconds
+        to it to come up with the next scheduled time.
+        """
+        return now
+
+    def __repr__(self):
+        return (
+            f"CeleryUserRateLimit(id_type={self.id_type!r}, "
+            f"id={self.id!r}, last_scheduled_time={self.last_scheduled_time!r})"
+        )
+
+
+class CeleryUserRateLimitStandard(CeleryUserRateLimit):
```

Comment:
This object hierarchy makes sense, of course. However, I wonder if it might be better suited for a manager, not a model? 

Here's why. According to SQLAlchemy documentation, when we have single table inheritance, a discriminator column is required ([ref](https://docs.sqlalchemy.org/en/14/orm/inheritance.html#single-table-inheritance)). Yes, here there is no distinction between the classes in terms of state (or mapped attributes) - it's just different implementation of the same method (which, again, make perfect sense). But I wonder whether the fact that these classes become mapped is not going to cause issues without specifying `polymorphic_identity` (I'm not sure). At the same time, adding that discriminator here seems silly and unnecessary. So, maybe move this logic into a manager (or managers), and keep the model focused on defining the mapped attributes? Again, I'm not at all sure about this, but it does feel like a simpler and safer alternative.